OG/OG/OG.g4

```diff
@@ -1,83 +1,216 @@
+mathExpression  : lhs=term op=Plus_Minus rhs=mathExpression        #infixAdditionExpr //operand står i midten
+                | child=term                                       #singleTermExpr
+                ;
+                
+term            : lhs=factor op=Mul_Div rhs=term                   #infixMultExpr              
+                | child=factor                                     #singleTermChild //Child er et vidtdækkende begreb
+                ;      
+factor          : lhs=atom pow='^' rhs=factor                      
+                | child=atom                                          
+                |'(' mathExpr=mathExpression ')'
+                ;
+
+atom            : funcCall=functionCall
+                | value=Number 
+                | xyValue=CoordinateXYValue 
+                | id=ID
+                ;
+               
+
+boolExpression  : id=ID                                                    #boolExprID
+                | value=BooleanValue                                          #boolExprTrueFalse
+                | funcCall=functionCall                                          #boolExprFuncCall
+                | lhs=mathExpression BoolOperator rhs=mathExpression    #boolExprMathComp
+                | lhs=boolExpression LogicOperator rhs=boolExpression   #boolExprBoolComp
+                | '!'boolExpr=boolExpression                                     #boolExprNotPrefix   
+                ;
 
-boolExpression  : ID |  BooleanValue |  functionCall 
-                | mathExpression BoolOperator mathExpression 
-                | boolExpression LogicOperator boolExpression 
-                | '!'boolExpression;  //Overvej function call
 
-//a == true er ikke gyldigt!
 
 
 //Commands:
 //Movement commands:
-command         : iterationCommand | movementCommand | drawCommand;
-
-movementCommand : (lineCommand | curveCommand)';';
-lineCommand     : 'line' '.' 'from' '(' (numberTuple | ID | pointReference)')' toCommand+;
-curveCommand    : 'curve' '.' 'withAngle' '(' mathExpression ')' '.' 'from' '('(numberTuple | ID)')' toCommand;
-toCommand       : '.to''(' (numberTuple | ID) ')';
-drawCommand     : ID';' | ID '.'fromCommand;
-fromCommand     :  'from' ( '('ID')' |  '(' numberTuple ')' | '(' StartPointReference ')' | '(' EndPointReference ')'| functionCall );
-
+command         : iterCmd=iterationCommand
+                | movementCmd=movementCommand
+                | drawCmd=drawCommand
+                ;
+
+movementCommand : lineCmd=lineCommand ';' 
+                | curveCmd=curveCommand';'
+                ;
+                
+lineCommand     : 'line' fromCmd=fromCommand  toCmds=toCommands;
+
+toCommands: toCmd=toCommand chainedToCmds=toCommands          #chainedToCommand
+          | toCmd=toCommand                                   #singleToCommand
+          ;
+
+   
+curveCommand    : 'curve''.''withAngle' '('angle=mathExpression ')'  fromCmd=fromCommand toCmd=toCommand;
```

Comment:
Kunne det give mening, at Command har en "Type" (line/curve) "Modifiers" (angle/mm.) "MovementCommands" (from/to)?

Så det bliver i den semantiske analyse, at vores TypeChecker tilføjer korrekte modifiers (aka. line.withAngle() er semantisk men ikke syntaktisk forkert). Gør det vel nemmere at udvide med andre modifikationer fremmadrettet?